Dockerfile.etcd

```diff
@@ -0,0 +1,4 @@
+ARG ETCD_VERSION
+
+# etcd uses gcr.io/etcd-development/etcd as a primary container registry, and quay.io/coreos/etcd as secondary.
+FROM gcr.io/etcd-development/etcd:$ETCD_VERSION
```

Comment:
Why have an intermediary docker container instead of just pulling from the upstream registry?

Reply:
Chatted a bit about this offline, I think we need a note in this file detailing why we need it instead of just pulling in from the third party repo (presumable for multi arch builds)

Reply:
[retagging seems to just publish your host architecture](https://pachyderm.slack.com/archives/CDWDPKPMY/p1663607841701349)